### ISSUE TITLE
can-import not working as expected

### ISSUE THREAD

**Issue by MarcGodard** [canjs/canjs#5187]
The following should only load one styles from my understanding, however all files are being loaded no matter the variable. It did work a few times so I am assuming some sort of race condition might be going on between the vm loading and the index.stache
```
  {{#switch (scope.vm.siteStyle)}}
    {{#case ('darkly')}}
      <can-import from="~/less/darkly/styles.less"></can-import>
    {{/case}}
    {{#case ('cosmo')}}
      <can-import from="~/less/cosmo/styles.less"></can-import>
    {{/case}}
    {{#case ('yeti')}}
      <can-import from="~/less/yeti/styles.less"></can-import>
    {{/case}}
  {{/switch}}
```
<img width="386" alt="Screen Shot 2019-08-04 at 9 21 56 PM" src="https://user-images.githubusercontent.com/5016637/62432610-aa806380-b6fe-11e9-9064-6efebbe8ce35.png">

I also tried the following with the same results:

```
  {{#eq (scope.vm.siteStyle, 'darkly')}}
    <can-import from="~/less/darkly/styles.less"></can-import>
  {{/eq}}
  {{#eq (scope.vm.siteStyle, 'cosmo')}}
    <can-import from="~/less/cosmo/styles.less"></can-import>
  {{/eq}}
  {{#eq (scope.vm.siteStyle, 'yeti')}}
    <can-import from="~/less/yeti/styles.less"></can-import>
  {{/eq}}
```

**Comment by MarcGodard** [comment]
@matthewp Can this be because steals needs to load everything for the build?

**Comment by matthewp** [comment]
Taking a look now.

**Comment by matthewp** [comment]
Yeah something is wrong, narrowing down the problem.

**Comment by matthewp** [comment]
The issue is that having a closing tag is not enough to make it be dynamic.  This is a bug. As a work around you can insert a space to trigger the parser to make it dynamic like so:

```handlebars
<can-import from="~/less/darkly/styles.less">&nbsp;</can-import>
```

__Or__ (and probably better anyways), use `can-dynamic-import`:

```handlebars
<can-dynamic-import from="~/less/darkly/styles.less"></can-dynamic-import>
```

Looking into what can be done to fix this though.

**Comment by matthewp** [comment]
Actually, just now realized that it's documented to work this way. If there's anything more than whitespace it is treated as a dynamic import.   https://canjs.com/doc/can-view-import.html#_can_importfrom__MODULE_NAME__content__can_import_

So tldr is to use can-dynamic-import.

**Comment by phillipskevin** [comment]
Would it be possible to add a warning if there is a closing `</can-import>` tag but it is being treated as a static import because of the content? This behavior is confusing, so it would be nice to guide people to use either
```html
<can-import ... />
```
or
```html
<can-dynamic-import ...>
...
</can-dynamic-import>
```

**Comment by matthewp** [comment]
Yeah, I think that's a reasonable thing to do. It *might* require changes to can-view-parser, however, in addition to can-stache-ast.
